kpi/models/asset_user_partial_permission.py

```diff
@@ -77,107 +153,8 @@ def update_partial_perms_to_include_implied(
-                    try:
-                        new_value = filter_[filter_key][in_op]
-                        current_values = current_filters[filter_key][in_op]
-                    except (KeyError, TypeError):
-                        pass
-                    else:
-                        new_partial_perm[0][filter_key][in_op] = list(
-                            set(current_values + new_value)
-                        )
-                        continue
-
-                # As said earlier, front end only supports `'_submitted_by'`
-                # filter, but many different and more complex filters could
-                # be used.
-                # If we reach these lines, it means conditions cannot be
-                # merged, so we concatenate then with an `OR` operator.
-                # Example:
-                # ```
-                # partial_perms = {
-                #   'view_submissions' : [{'_submitted_by': 'johndoe'}],
-                #   'delete_submissions':  [
-                #       {'_submission_date': {'$lte': '2021-01-01'},
-                #       {'_submission_date': {'$gte': '2020-01-01'}
-                #   ]
-                # }
-                # ```
-                # should give
-                # ```
-                # new_partial_perms = {
-                #   'view_submissions' : [
-                #           [{'_submitted_by': 'johndoe'}],
-                #           [
-                #               {'_submission_date': {'$lte': '2021-01-01'},
-                #               {'_submission_date': {'$gte': '2020-01-01'}
-                #           ]
-                #   },
-                #   'delete_submissions':  [
-                #       {'_submission_date': {'$lte': '2021-01-01'},
-                #       {'_submission_date': {'$gte': '2020-01-01'}
-                #   ]
-                # }
-
-                # To avoid more complexity (and different syntax than
-                # trivial case), we delegate to MongoHelper the task to join
-                # lists with the `$or` operator.
-                try:
-                    new_partial_perm = new_partial_perms[implied_perm][0]
-                except IndexError:
-                    # If we get an IndexError, implied permission does not
-                    # belong to current assignment. Let's copy the filters
-                    #
-                    new_partial_perms[implied_perm] = filters
-                else:
-                    if not isinstance(new_partial_perm, list):
-                        new_partial_perms[implied_perm] = [
-                            filters,
-                            new_partial_perms[implied_perm],
-                        ]
-                    else:
-                        new_partial_perms[implied_perm].append(filters)
+                add_implied_permission(
+                    new_partial_perms, partial_perm, implied_perm
+                )
```

Comment:
My own preference: I would have kept `add_implied_permission` as a static "protected" method of the current class.